### PR TITLE
Add tutorial for constrained problems

### PR DESCRIPTION
**Pull request by y0z**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- [x] I agree to the contributor agreements.

> [!TIP]
> Please follow the [Quick TODO list](https://github.com/optuna/optunahub-registry/tree/main?tab=readme-ov-file#quick-todo-list-towards-contribution) to smoothly merge your PR.

## Motivation & Description of the changes

Add a tutorial to make constrained benchmark problems.
**Before merging this PR, please merge https://github.com/optuna/optunahub/pull/75.**

### CODE REVIEW

File: recipes/007_benchmarks_advanced.py [hunk 1]
```diff
@@ -63,6 +64,36 @@ def evaluate(self, params: dict[str, float]) -> float:
...
 study.optimize(dynamic_problem, n_trials=20)
 
+
+###################################################################################################
+# Implementing a problem with constraints
+# -------------------------------------------------
+# Here, let's implement a problem with constraints.
+# To implement a problem with constraints, you need to inherit ``ConstrainedMixin`` class in addition to ``BaseProblem`` and implement the ``evaluate_constraints`` method.
+# The ``evaluate_constraints`` method evaluates the constraint functions given a dictionary of input parameters and returns a list of constraint values.
+# Then, ``ConstrainedMixin`` internally defines the ``constraints_func`` method properly for Optuna samplers to handle constraints.
```
Comment (nabenabe0928) [inline recipes/007_benchmarks_advanced.py:74]:
```suggestion
# Then, ``ConstrainedMixin`` internally defines the ``constraints_func`` method for Optuna samplers.
```

File: recipes/007_benchmarks_advanced.py [hunk 2]
```diff
@@ -63,6 +64,36 @@ def evaluate(self, params: dict[str, float]) -> float:
...
+            return [c0]
+        else:
+            y = params["y"]
+            c1 = x + y - 3
+            return [c0, c1]
+
+
+###################################################################################################
+# Then, you can optimize the problem with Optuna as usual.
+# Don't forget to set the `constraints_func` argument of the sampler to the `constraints_func` of the problem.
```
Comment (nabenabe0928) [inline recipes/007_benchmarks_advanced.py:89]:
```suggestion
# Don't forget to set the `constraints_func` argument to the sampler to use.
```

File: recipes/007_benchmarks_advanced.py [hunk 3]
```diff
@@ -63,6 +64,36 @@ def evaluate(self, params: dict[str, float]) -> float:
...
+# Here, let's implement a problem with constraints.
+# To implement a problem with constraints, you need to inherit ``ConstrainedMixin`` class in addition to ``BaseProblem`` and implement the ``evaluate_constraints`` method.
+# The ``evaluate_constraints`` method evaluates the constraint functions given a dictionary of input parameters and returns a list of constraint values.
+# Then, ``ConstrainedMixin`` internally defines the ``constraints_func`` method properly for Optuna samplers to handle constraints.
+class ConstrainedProblem(ConstrainedMixin, DynamicProblem):
+    def evaluate_constraints(self, params: dict[str, float]) -> list[float]:
+        x = params["x"]
+        c0 = x - 2
+        if "y" not in params:
+            return [c0]
```
Comment (nabenabe0928) [inline recipes/007_benchmarks_advanced.py:80]:
```suggestion
            c1 = 0.0  # c1 <= 0, so c1 is satisfied in this case.
            return c0, c1
```

File: recipes/007_benchmarks_advanced.py [hunk 4]
```diff
@@ -63,6 +64,36 @@ def evaluate(self, params: dict[str, float]) -> float:
...
+
+###################################################################################################
+# Implementing a problem with constraints
+# -------------------------------------------------
+# Here, let's implement a problem with constraints.
+# To implement a problem with constraints, you need to inherit ``ConstrainedMixin`` class in addition to ``BaseProblem`` and implement the ``evaluate_constraints`` method.
+# The ``evaluate_constraints`` method evaluates the constraint functions given a dictionary of input parameters and returns a list of constraint values.
+# Then, ``ConstrainedMixin`` internally defines the ``constraints_func`` method properly for Optuna samplers to handle constraints.
+class ConstrainedProblem(ConstrainedMixin, DynamicProblem):
+    def evaluate_constraints(self, params: dict[str, float]) -> list[float]:
```
Comment (nabenabe0928) [inline recipes/007_benchmarks_advanced.py:76]:
```suggestion
    def evaluate_constraints(self, params: dict[str, float]) -> tuple[float]:
```

File: recipes/007_benchmarks_advanced.py [hunk 5]
```diff
@@ -63,6 +64,36 @@ def evaluate(self, params: dict[str, float]) -> float:
...
+        x = params["x"]
+        c0 = x - 2
+        if "y" not in params:
+            return [c0]
+        else:
+            y = params["y"]
+            c1 = x + y - 3
+            return [c0, c1]
```
Comment (nabenabe0928) [inline recipes/007_benchmarks_advanced.py:84]:
```suggestion
            return c0, c1
```

**Review by nabenabe0928 — COMMENTED** [review]
I left some comments!

**Comment by y0z** [comment]
I fixed the code and it now passes CI.

**Review by nabenabe0928 — APPROVED** [review]

**Pull request merged**